ml-proto/host/lexer.mll

```diff
@@ -88,18 +88,19 @@ let escape = ['n''t''\\''\'''\"']
 let character =
   [^'"''\\''\x00'-'\x1f''\x7f'] | '\\'escape | '\\'hexdigit hexdigit
 
-let sign = ('+' | '-')?
-let num = sign digit+
-let hexnum = sign "0x" hexdigit+
-let int = num | hexnum
+let sign = ('+' | '-')
+let num = digit+
+let hexnum = "0x" hexdigit+
+let nat = num | hexnum
+let int = sign nat
```

Comment:
@sunfishcode, tokens necessarily define disjoint lexical classes, so to carve out a subclass you need to union in the parser.

Yes, `nat` is "natural".